src/Shell/Process.php

```diff
@@ -487,6 +590,10 @@ public function read()
      */
     protected function readStream($id)
     {
+        if (isset($this->descriptorspec[$id]) && (is_resource($this->descriptorspec[$id]) || (isset($this->descriptorspec[$id][0]) && $this->descriptorspec[$id][0] == 'file'))) {
+            return '';
```

Comment:
I'd prefer this be made into a function - a simple function name is quicker to understand than an if with a multipart or. Another benefit is extensibility - definition of what readable is can be change. 

The multiple conditions can be broke into simple checks with early returns.

Reply:
Yes the expression is very hard to read, I see no use for methods but [I created variables](https://github.com/nickel715/shell-command/commit/9bd95a0f822c01af4dbb931f130993ef8c07f040) to improve readability